src/evolution_metrics.jl

```diff
@@ -0,0 +1,151 @@
+struct IndicatorEvolutionResults{T<:AbstractFloat}
+    t_indicator::AbstractVector{T}
+    x_indicator::AbstractVector{T}
+    t_evolution::AbstractVector{T}
+    x_evolution::AbstractVector{T}
+    surr_x_evolution::AbstractMatrix{T}
```

Comment:
All of these fields are abstract types (bad for performance). Why thought? Is there a specific reason we don't just have `Vector{T}`? We are computing these quantities, right?

Reply:
Correct, fixed!